collect-logs

```diff
@@ -267,18 +268,63 @@ def get_units(juju, status=None):
                  "is set.")
     return juju_units
 
-def _create_ps_output_file(juju, unit):
-    """List running processes and redirect them to a file."""
-    log.info("Collecting ps output on unit {}".format(unit.name))
-    ps_cmd = "ps fauxww | sudo tee /var/log/ps-fauxww.txt"
-    args = juju.ssh_args(unit, ps_cmd)
+def get_hosts(juju, status=None):
+    """Return a list of machine hosts (not lxds)."""
+    if status is None:
+        status = juju_status(juju)
+    if "machines" not in status:
+        sys.exit("ERROR, no machines found. Make sure the right juju "
+                 "environment is set.")
+    juju_hosts = []
+    for name, machine in status["machines"].items():
+        juju_hosts.append(JujuHost(name, machine["dns-name"]))
+    return juju_hosts
+
+
+def _run_cmd(juju, unit, cmd, message):
+    """Helper method for running a command on a juju unit."""
+    args = juju.ssh_args(unit, cmd)
     try:
         check_output(args, stderr=STDOUT, env=juju.env)
     except CalledProcessError as e:
-        log.warning(
-            "Failed to collect running processes on unit {}".format(unit.name))
+        log.warning(message)
         log.warning(e.output)
         log.warning(e.returncode)
+        return False
+    return True
+
+
+def _create_ps_output_file(juju, unit):
+    """List running processes and redirect them to a file."""
+    log.info("Collecting ps output on unit {}".format(unit.name))
+    message = "Failed to collect running processes on unit {}".format(unit.name)
+    ps_cmd = "ps fauxww | sudo tee /var/log/ps-fauxww.txt"
+    _run_cmd(juju, unit, ps_cmd, message)
+
+
+def _create_ps_mem_output_file(juju, unit):
+    """Gather the aggregate memory footprint of each process using ps_mem."""
+    log.info("Collecting ps_mem output on unit {}".format(unit.name))
+
+    # ps_mem requries python 2, make sure it is installed
+    apt_cmd = "if ! python -V; then sudo apt-get install -y python; fi"
+    message = "Failed to install python on unit {}".format(unit.name)
+    if not _run_cmd(juju, unit, apt_cmd, message):
+        return
+
+    # Fetch the ps_mem utility from the github repo
+    ps_mem_repo = "https://github.com/fginther/ps_mem.git"
```

Comment:
There is now an `upload_ps_mem()` which does the git clone on the host system executing `collect-logs`. This is called twice, once for the other host 0 and again to upload ps_mem onto the landscape-server/0 node for use by the inner model.